web/src/enterprise/campaigns/detail/changesets/ChangesetFilterRow.tsx

```diff
@@ -0,0 +1,189 @@
+}) => {
+    const searchParameters = new URLSearchParams(location.search)
+    const [uiState, setUIState] = useState<ChangesetUIState | undefined>(() => {
+        const value = searchParameters.get('status')
+        return value && isValidChangesetUIState(value) ? value : undefined
+    })
+    const [reviewState, setReviewState] = useState<ChangesetReviewState | undefined>(() => {
+        const value = searchParameters.get('review_state')
+        return value && isValidChangesetReviewState(value) ? value : undefined
+    })
+    const [checkState, setCheckState] = useState<ChangesetCheckState | undefined>(() => {
+        const value = searchParameters.get('check_state')
+        return value && isValidChangesetCheckState(value) ? value : undefined
+    })
+    useEffect(() => {
+        const searchParameters = new URLSearchParams(location.search)
+        if (uiState) {
+            searchParameters.set('status', uiState)
+        } else {
+            searchParameters.delete('status')
+        }
+        if (reviewState) {
+            searchParameters.set('review_state', reviewState)
+        } else {
+            searchParameters.delete('review_state')
+        }
+        if (checkState) {
+            searchParameters.set('check_state', checkState)
+        } else {
+            searchParameters.delete('check_state')
+        }
+        history.replace({ ...location, search: searchParameters.toString() })
+        // Update the filters in the parent component.
+        onFiltersChange({
+            ...(uiState
+                ? changesetUIStateToChangesetFilters(uiState)
+                : {
+                      reconcilerState: null,
+                      publicationState: null,
+                      externalState: null,
+                  }),
+            reviewState: reviewState || null,
+            checkState: checkState || null,
+        })
+        // We cannot depend on the history, since it's modified by this hook and that would cause an infinite render loop.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [uiState, reviewState, checkState])
+    return (
+        <div className="form-inline m-0">
+            <ChangesetFilter<ChangesetUIState>
+                values={Object.values(ChangesetUIState)}
+                label="Status"
+                selected={uiState}
+                onChange={setUIState}
+                className="mr-2"
+            />
+            <ChangesetFilter<ChangesetReviewState>
+                values={Object.values(ChangesetReviewState)}
+                label="Review state"
+                selected={reviewState}
+                onChange={setReviewState}
+                className="mr-2"
+            />
+            <ChangesetFilter<ChangesetCheckState>
+                values={Object.values(ChangesetCheckState)}
+                label="Check state"
+                selected={checkState}
+                onChange={setCheckState}
+            />
```

Comment:
Not sure whether @rrhyne agrees with that, but the columns in the list are: `status`, ..., `checkState`, `reviewState`. The order of the filters, *I* think, should match that:

```suggestion
            <ChangesetFilter<ChangesetCheckState>
                values={Object.values(ChangesetCheckState)}
                label="Check state"
                selected={checkState}
                onChange={setCheckState}
            />
            <ChangesetFilter<ChangesetReviewState>
                values={Object.values(ChangesetReviewState)}
                label="Review state"
                selected={reviewState}
                onChange={setReviewState}
                className="mr-2"
            />
```

Reply:
Agree, changed it